lib/python/qmk/cli/new/keyboard.py

```diff
@@ -0,0 +1,182 @@
+        keyboard_name = input("\n    Name: ")
+
+
+    if cli.args.microcontroller:
+        mcu = cli.args.microcontroller.lower()
+        for key, value in enumerate(mcus):
+            if re.match( r'^' + value[0] + '$', mcu, re.I ):
+                arch = value[1]
+                break
+        else:
+            cli.log.error(mcu + " is not a valid microcontroller option.")
+            options = [];
+            for i in range(0, len(mcus)):
+                options.append( mcus[i][0] )
+            print("  Valid Options: ", ", ".join(options) )
+            exit(1)
+    else:
+    # Ask what microcontroller is being used
+        print("\n** Select the microcontroller used: **\n")
+        print_microcontrollers()
+
+        mcu = int(input("\n    Microcontroller: (1-" + str(len(mcus)) + "): "))
+        # user-facing text is 1-indexed, but data is 0-indexed internally
+        mcu = mcu - 1
+        arch = mcus[mcu][1]
+        mcu = mcus[mcu][0]
+        mcu = mcu.lower()
+    # Set the path to the MCU architecture's template files
+    template_arch_path = Path(template_root_path) / arch
+
+
+    # (Erovia) introducing the user.name variable would probably be better
+    if cli.args.username:
+        user_name = cli.args.username
+    elif cli.config.user.keymap:
+        user_name = cli.config.user.keymap
+    else:
+        # Ask the user for their GitHub username
+        print("""\n** What is your GitHub username? **
+
+        This value will be pasted into the new files in copyright headers, and used
+        as the listed Keyboard Maintainer in the readme file.""")
+        user_name = input("\n    GitHub Username: ")
+
+    # generate keyboard paths
+    keyboard_path = Path("keyboards") / keyboard
+
+    # check directories
+    if Path.exists(keyboard_path):
+        cli.log.error('Keyboard %s already exists!', keyboard)
+        exit(1)
+    elif keyboard.count('/') > 0:
+        parent_directory = "/".join(keyboard.split('/')[0:len(keyboard.split('/')) - 1] )
+        Path(parent_directory).mkdir(parents=True, exist_ok=True)
+    final_directory = keyboard.split('/')[-1]
+
+    # create user directory with default keyboard files
+    shutil.copytree(template_base_path, keyboard_path, symlinks=True)
+    kb_c = Path(keyboard_path) / "keyboard.c"
+    kb_h = Path(keyboard_path) / "keyboard.h"
+    kb_c.rename( Path(keyboard_path) / Path(final_directory + ".c") )
+    kb_h.rename( Path(keyboard_path) / Path(final_directory + ".h") )
+
+    # copy architecture files
+    shutil.copy(Path(template_arch_path) / "config.h", keyboard_path)
+    shutil.copy(Path(template_arch_path) / "readme.md", keyboard_path)
+    shutil.copy(Path(template_arch_path) / "rules.mk", keyboard_path)
+    if ( arch == "ps2avrgb" ):
+        # only ps2avrgb keyboards require this file
+        shutil.copy(Path(template_arch_path) / "usbconfig.h", keyboard_path)
+    if ( arch == "stm32" ):
```

Comment:
No parens around if's unless needed for math purposes.

```suggestion
    if arch == "stm32":
```